### PR TITLE
docs: Add ingress TLS cipher and version documentation

### PR DESCRIPTION
**Pull request by blake**
Document the new TLS cipher and version parameters that were added to ingress gateways in #11576.

### CODE REVIEW

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 1]
```diff
@@ -1138,6 +1138,45 @@ You can specify the following parameters to configure ingress gateway configurat
...
+                            TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA<br>
+                            TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA<br>
+                            TLS_RSA_WITH_AES_128_GCM_SHA256<br>
+                            TLS_RSA_WITH_AES_128_CBC_SHA<br>
+                            TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384<br>
+                            TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384<br>
+                            TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA<br>
+                            TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA<br>
+                            TLS_RSA_WITH_AES_256_GCM_SHA384<br>
+                            TLS_RSA_WITH_AES_256_CBC_SHA\`\`\``
```
Comment (mikemorris) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1201]:
```suggestion
                            [default server cipher list](https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/transport_sockets/tls/v3/common.proto#envoy-v3-api-field-extensions-transport-sockets-tls-v3-tlsparameters-cipher-suites"). The list of supported cipher suites can seen in [`types/tls.go`](https://github.com/hashicorp/consul/blob/main/types/tls.go#L154-L169) and is dependent on underlying support in Envoy - future releases of Envoy may remove currently-supported but insecure cipher suites, and future releases of Consul may add new supported cipher suites if any are added to Envoy.
```

Comment (mikemorris) [reply website/content/docs/connect/config-entries/ingress-gateway.mdx:1201]: Linking to the actual implementation source instead could help ensure this content remains current, and is similar to how the Agent TLS config docs handle this https://www.consul.io/docs/agent/options#tls_cipher_suites

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 2]
```diff
@@ -1138,6 +1138,45 @@ You can specify the following parameters to configure ingress gateway configurat
                   "Set this configuration to `true` to enable built-in TLS for this listener.<br><br>If TLS is enabled, then each host defined in the `hosts` field will be added as a DNSSAN to the gateway's x509 certificate. Note that even hosts from other listeners with TLS disabled will be added.",
               },
             },
+            {
+              name: 'TLSMinVersion',
+              type: 'string: ""',
+              description: "Set the minimum TLS version supported for this listener. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, or `TLSv1_3`.",
```
Comment (mikemorris) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1167]:
```suggestion
              description: "Set the minimum TLS version supported for this listener. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, or `TLSv1_3`. If unspecified, Envoy v1.22.0 and newer [will default to TLS 1.2 as a min version](https://github.com/envoyproxy/envoy/pull/19330), while older releases of Envoy default to TLS 1.0.",
```

**Review by mikemorris — APPROVED** [review]
I think this looks good, thanks for writing this up! Added the corresponding gateway config options, and tweaked some phrasing around default behavior and supported values.

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 3]
```diff
@@ -979,6 +979,36 @@ You can specify the following parameters to configure ingress gateway configurat
...
+            yaml:
+              "Set the maximum TLS version supported for the gateway's listeners. Must be greater than or equal to `tls_min_version`. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, or `TLSv1_3`." ,
+          },
+        },
+        {
+          name: 'CipherSuites',
+          type: 'array<string>: <optional>',
+          description: `A list of TLS cipher suites to support when negotiating
+                        connections using TLS 1.2 or earlier. If unspecified,
+                        Envoy will use a
```
Comment (mikemorris) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1002]:
```suggestion
          description: `Set the default list of TLS cipher suites for the gateway's
                        listeners to support when negotiating connections using
                        TLS 1.2 or earlier. If unspecified, Envoy will use a
```

File: website/content/docs/connect/config-entries/ingress-gateway.mdx [hunk 4]
```diff
@@ -1133,11 +1163,40 @@ You can specify the following parameters to configure ingress gateway configurat
...
+                hcl:
+                  "Set the maximum TLS version supported for this listener. Must be greater than or equal to `TLSMinVersion`. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, or `TLSv1_3`." ,
+                yaml:
+                  "Set the maximum TLS version supported for this listener. Must be greater than or equal to `tls_min_version`. One of `TLS_AUTO`, `TLSv1_0`, `TLSv1_1`, `TLSv1_2`, or `TLSv1_3`." ,
+              },
+            },
+            {
+              name: 'CipherSuites',
+              type: 'array<string>: <optional>',
+              description: `A list of TLS cipher suites to support when negotiating
```
Comment (mikemorris) [inline website/content/docs/connect/config-entries/ingress-gateway.mdx:1189]:
```suggestion
              description: `Set the list of TLS cipher suites to support when negotiating
```

**Review by mikemorris — APPROVED** [review]
Added a few more tweaks to clarify that most gateway config can be superseded by listener-specific config.

**Review by karl-cardenas-coding — APPROVED** [review]
lgtm

**Comment by hc-github-team-consul-core** [comment]
:cherries: If backport labels were added before merging, cherry-picking will start automatically.

To retroactively trigger a backport after merging, add backport labels and re-run https://circleci.com/gh/hashicorp/consul/566092.

**Comment by hc-github-team-consul-core** [comment]
:cherries::white_check_mark: Cherry pick of commit a3ad4be429ab004600f99ad53afee28e3843e1a8 onto `stable-website` succeeded!

**Pull request merged**